src/cms/views/linkcheck/linkcheck.py

```diff
@@ -0,0 +1,106 @@
+from django.db.models import Q, OuterRef, Subquery
+from django.urls import reverse
+from django.views.generic import ListView
+from django.views.generic.base import RedirectView
+
+from linkcheck.models import Link
+from backend.settings import PER_PAGE
+
+from ...models.pages.page_translation import PageTranslation
+from ...models.events.event_translation import EventTranslation
+from ...models.pois.poi_translation import POITranslation
+
+
+# pylint: disable=too-many-ancestors
+class LinkListView(ListView):
+    """
+    View for retrieving a list of links grouped by their state
+    """
+
+    template_name = "linkcheck/links_by_filter.html"
+    context_object_name = "filtered_links"
+    paginate_by = PER_PAGE
+    extra_context = {"current_menu_item": "linkcheck"}
+
+    def get_context_data(self, **kwargs):
+        """
+        Appends additional context like the captured variable in the url
+
+        :return: The context rendered in the template
+        :rtype: dict
+        """
+        context = super().get_context_data(**kwargs)
+        context.update(
+            {
+                **self.kwargs,
+                "current_menu_item": "linkcheck",
+            }
+        )
+        return context
+
```

Comment:
```suggestion
```
Since you don't really have dynamic context data, you don't need the function for it. You can just put everything in the `extra_context` variable. The `current_menu_item` is contained in it anyway and the other variables can be added to it as well.